Implement landing page for submission portal

eecavanna opened the pull request:
In this branch, I added some new sections to the submission portal landing page.

### `TitleBanner.vue`

This component contains a page title and subtitle overlaying a banner image. The component was modeled after a header on the NMDC website.

![image](https://github.com/microbiomedata/nmdc-server/assets/134325062/8c66e917-e928-494c-80de-64f4b18d2f7d)

> Note: The banner image is a 470 KB PNG file (i.e. `submission-portal-hero.png`). I failed to find an SVG version and don't think one exists. Given that the image contains a color gradient, I'm wondering whether a JPG version would be significantly smaller while being insignificantly lower quality.

### `IconBar.vue`

![image](https://github.com/microbiomedata/nmdc-server/assets/134325062/23d010f9-e8df-4513-85ef-ca26f7050373)

This component contains a row of three icon-title pairs, each pair linking to some documentation. The icons belong to the Material Design Icons library, which is used elsewhere in the code base.

### `IntroBlurb.vue`

![image](https://github.com/microbiomedata/nmdc-server/assets/134325062/e80c999c-ee3c-44f6-92da-8945f3506b56)

This component contains three paragraphs that introduce the submission portal. In this component's name, I used "blurb" to refer to the entire set of text.

This is only visible when the user is either (a) logged out; or (b) logged in, but does not have any submissions.

### `LoginPrompt.vue`

![image](https://github.com/microbiomedata/nmdc-server/assets/134325062/3c9bc52e-dbfd-4950-99e7-28f96765540f)

This component contains a plainly-styled "alert" inviting the visitor to log into ORCiD; or create an ORCiD if they don't have one. This takes the place of the previous, loose login message.

### Components in context

Here are links to screenshots showing the components in context.

- Visitors that are not logged in will see [this](https://github.com/microbiomedata/nmdc-server/issues/1144#issuecomment-1975072897).
- Visitors that are logged in and have no submissions will see [this](https://github.com/microbiomedata/nmdc-server/issues/1144#issuecomment-1975074277).
- Visitors that are logged in and have submissions will see [this](https://github.com/microbiomedata/nmdc-server/issues/1144#issuecomment-1975073107).

---

There is one remaining task, but I don't want to hold up the review(s) for it. That remaining task is to finalize the link to the schema documentation. I will pursue that after requesting reviews as I don't expect it to have an impact on the reviews (it'll be a one-line change).

eecavanna commented on web/src/views/SubmissionPortal/Components/IconBar.vue:
```diff
@@ -0,0 +1,101 @@
+        </a>
+      </v-col>
+      <v-col>
+        <!-- TODO: Verify that this is the correct `href`, given recent decision about documentation duplication. -->
```
TODO: @eecavanna

eecavanna commented on web/src/views/SubmissionPortal/Components/IconBar.vue:
```diff
@@ -0,0 +1,101 @@
+        </a>
+      </v-col>
+      <v-col>
+        <!-- TODO: Verify that this is the correct `href`, given recent decision about documentation duplication. -->
+        <a
+          target="_blank"
+          title="View the metadata schema"
+          href="https://nmdc-documentation.readthedocs.io/en/latest/reference/metadata/combined_schema_docs.html"
+          class="text-decoration-none"
+        >
```
TODO: @eecavanna

eecavanna replied:
I'll update it to point to: https://w3id.org/nmdc/about

marySalvi commented on web/src/views/SubmissionPortal/Components/IntroBlurb.vue:
```diff
@@ -0,0 +1,59 @@
+} from '@vue/composition-api';
+
+export default defineComponent({
+  components: {},
+  setup() {
+    return {};
+  },
```
remove

eecavanna replied:
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

marySalvi commented on web/src/views/SubmissionPortal/Components/IntroBlurb.vue:
```diff
@@ -0,0 +1,59 @@
+          The Submission Portal leverages validation functions of the
+          <a
+            href="https://github.com/cidgoh/DataHarmonizer"
+            target="_blank"
```
add `rel = noreferrer nopener` for extra security.

eecavanna replied:
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

marySalvi commented on web/src/views/SubmissionPortal/Components/IntroBlurb.vue:
```diff
@@ -0,0 +1,59 @@
+          tool to check entered metadata values against the standards in the
+          <a
+            href="https://github.com/microbiomedata/nmdc-schema"
+            target="_blank"
```
add `rel = noreferrer nopener` for extra security.

eecavanna replied:
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

marySalvi commented on web/src/views/SubmissionPortal/Components/LoginPrompt.vue:
```diff
@@ -0,0 +1,56 @@
+import { defineComponent } from '@vue/composition-api';
+
+export default defineComponent({
+  props: {},
```
remove

eecavanna replied:
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

marySalvi commented on web/src/views/SubmissionPortal/Components/LoginPrompt.vue:
```diff
@@ -0,0 +1,56 @@
+            Login with your ORCiD
+            to access the Submission Portal.
+          </p>
+          <p class="body-2 mb-0 text--disabled">
+            Don't have an ORCiD? Individuals can
+            <a
+              href="https://support.orcid.org/hc/en-us/articles/360006973953"
+              target="_blank"
+              title="Learn about registering an ORCiD"
+              class="text--disabled"
+            >get one for free</a>.
+          </p>
```
I am concerned about the readability of this text. The contrast from the background is low.

eecavanna replied:
Thank you for bringing this to my attention, @marySalvi!

I plugged in the colors into [an online tool](https://webaim.org/resources/contrastchecker/) and got "failing grades":

![image](https://github.com/microbiomedata/nmdc-server/assets/134325062/2b7a72c1-a73f-45b2-b5ba-92afab62b8a2)

I will increase the contrast (TBD).

eecavanna replied:
I changed it to black in commit 6865d7c86fe6bb33a5c764a6645b0537b32be8a0. That online tool shows "passing" all across the board, for black (on gray).

marySalvi commented on web/src/views/SubmissionPortal/Components/LoginPrompt.vue:
```diff
@@ -0,0 +1,56 @@
+            Don't have an ORCiD? Individuals can
+            <a
+              href="https://support.orcid.org/hc/en-us/articles/360006973953"
+              target="_blank"
```
add rel = noreferrer nopener for extra security.

eecavanna replied:
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

marySalvi commented on web/src/views/SubmissionPortal/Components/LoginPrompt.vue:
```diff
@@ -0,0 +1,56 @@
+          </p>
+        </v-col>
+        <v-col class="flex-grow-0">
+          <!-- Note: This button is a simplified version of: `web/src/components/Presentation/AuthButton.vue` -->
```
I don't think we'll need this comment long term, can probably remove it.

eecavanna replied:
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

marySalvi commented on web/src/views/SubmissionPortal/Components/LoginPrompt.vue:
```diff
@@ -0,0 +1,56 @@
+<template>
+  <v-alert
+    class="mb-0"
+    text="text"
```
text is a boolean prop so no need to set a value

eecavanna replied:
Thanks for bringing that to my attention! Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

marySalvi commented on web/src/views/SubmissionPortal/Components/TitleBanner.vue:
```diff
@@ -0,0 +1,27 @@
+
+<template>
+  <v-container>
+    <v-row :style="{ backgroundImage: 'url(' + require('@/assets/submission-portal-hero.png') + ')', backgroundSize: 'cover' }">
```
Hooray! Setting an image to background image when it is not info bearing is excellent accessibility practice. :)

marySalvi reviewed (changes requested):
Some minor changes but overall looks great!

eecavanna commented on web/src/views/SubmissionPortal/Components/IconBar.vue:
```diff
@@ -0,0 +1,101 @@
+  components: {},
+  setup() {
+    return {};
+  },
```
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

eecavanna commented on web/src/views/SubmissionPortal/Components/IconBar.vue:
```diff
@@ -0,0 +1,101 @@
+      <v-col>
+        <!-- TODO: Verify that this is the correct `href`, given recent decision about documentation duplication. -->
+        <a
+          target="_blank"
```
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

eecavanna commented on web/src/views/SubmissionPortal/Components/IconBar.vue:
```diff
@@ -0,0 +1,101 @@
+      </v-col>
+      <v-col>
+        <a
+          target="_blank"
```
Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

eecavanna commented on web/src/views/SubmissionPortal/Components/IconBar.vue:
```diff
@@ -0,0 +1,101 @@
+    <v-row>
+      <v-col>
+        <a
+          target="_blank"
```
Thank you for providing the "fix" and a link to some additional information! Done in commit 794d5b95eef58a33da7c4cbe32cc04e40c6c2778.

eecavanna commented:
I have addressed all the review feedback.

FYI: I am still waiting on guidance from a team member regarding where one of the links will point. I'll reach out to her again about it.

eecavanna commented:
I'm ready for this to be merged in (assuming nobody objects to its contents).

naglepuff reviewed (changes requested):
Sorry to swoop in last minute like this but the "i" in ORCID should be capitalized. The "i" in ORCID iD is lowercase.

eecavanna commented on web/src/views/SubmissionPortal/Components/LoginPrompt.vue:
```diff
@@ -0,0 +1,54 @@
+      <v-row align="center">
+        <v-col class="justify-content-center">
+          <p class="body-1">
+            Login with your ORCiD
```
I'll make this change now.
